File: FantasyRadio/src/main/java/ru/sigil/log/DefaultLogger.java
package ru.sigil.log;

        import android.util.Log;

/**
 * User: Nikita
 * Date: 21.03.12
 */
public class DefaultLogger implements ICustomLogger {

    @Override
    public synchronized void handle(LogLevel logLevel, String tag, String component, String message, Exception e) {
        switch (logLevel) {
            case DEBUG:
                Log.d(tag, "(" + component + ") " + message, e);
                break;
            case WARNING:
                Log.w(tag, "(" + component + ") " + message, e);
                break;
            case ERROR:
                Log.e(tag, "(" + component + ") " + message, e);
                break;
        }
    }
}

File: FantasyRadio/src/main/java/ru/sigil/log/DefaultLogger.kt
package ru.sigil.log

import android.util.Log

/**
 * Created by namelessone
 * on 08.12.18.
 */
class DefaultLogger : ICustomLogger {
    @Synchronized
    override fun handle(logLevel: LogLevel, tag: String, component: String, message: String, e: Exception) {
        when (logLevel) {
            LogLevel.DEBUG -> Log.d(tag, "($component) $message", e)
            LogLevel.WARNING -> Log.w(tag, "($component) $message", e)
            LogLevel.ERROR -> Log.e(tag, "($component) $message", e)
        }
    }
}